File: app/src/main/java/de/bikebean/app/db/settings/settings/replace_if_newer_settings/Wifi.java
package de.bikebean.app.db.settings.settings.replace_if_newer_settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.settings.settings.ReplaceIfNewerSetting;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.sms.SmsFactory;
import de.bikebean.app.db.state.State;
import de.bikebean.app.db.state.StateFactory;

public class Wifi extends ReplaceIfNewerSetting {

    public static final boolean INITIAL_WIFI = false;

    private static final @NonNull State.KEY key = State.KEY.WIFI;

    public Wifi(final @NonNull Sms sms, final @NonNull WifiGetter wifiGetter) {
        super(sms,
                StateFactory.createNumberState(
                        sms, key, wifiGetter.getWifi() ? 1.0 : 0.0, State.STATUS.CONFIRMED
                )
        );
    }

    public Wifi(boolean wifi, final @NonNull Sms sms) {
        super(sms,
                StateFactory.createNumberState(
                        sms, key, wifi ? 1.0 : 0.0,
                        State.STATUS.CONFIRMED
                )
        );
    }

    public Wifi() {
        super(SmsFactory.createNullSms(),
                StateFactory.createNumberState(
                        SmsFactory.createNullSms(), key, INITIAL_WIFI ? 1.0 : 0.0,
                        State.STATUS.CONFIRMED
                )
        );
    }

    public interface WifiGetter {
        boolean getWifi();
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/replace_if_newer_settings/Wifi.kt
package de.bikebean.app.db.settings.settings.replace_if_newer_settings

import de.bikebean.app.db.settings.settings.ReplaceIfNewerSetting
import de.bikebean.app.db.sms.Sms
import de.bikebean.app.db.sms.SmsFactory
import de.bikebean.app.db.state.State
import de.bikebean.app.db.state.State.KEY
import de.bikebean.app.db.state.StateFactory

class Wifi : ReplaceIfNewerSetting {

    constructor(
            wifi: Boolean, sms: Sms
    ) : super(StateFactory.createNumberState(
            sms, key, when {
                wifi -> 1.0
                else -> 0.0
            }, status
    ))

    constructor(
            sms: Sms, wifiGetter: () -> Boolean
    ) : this(wifiGetter(), sms)

    constructor() : super(StateFactory.createNumberState(
            SmsFactory.createNullSms(), key, 0.0, status
    ))

    companion object {
        const val INITIAL_WIFI = false
        private val status = State.STATUS.CONFIRMED
        private val key = KEY.WIFI
    }
}